### ISSUE TITLE
URI.parser deprecation warning for Rails 6.1

### ISSUE THREAD

**Issue by key88sf** [Shopify/shopify-api-ruby#826]
Getting this deprecation warning when trying to set `ShopifyAPI::Base.site` in a standard Rails initializer:

```
WARN -- : DEPRECATION WARNING: URI.parser is deprecated and will be removed in Rails 6.2. Use `URI::DEFAULT_PARSER` instead.
```

Are there plans to make the shopify_api gem compliant with Rails 6.2+ without errors?

**Comment by key88sf** [comment]
NOTE: This is due to the gem requiring ActiveResource < 6.0.0  (so it's using 5.1.1).

**Comment by paulomarg** [comment]
Hi @key88sf, thanks for reporting this, we will look into updating ActiveResource past v6.

**Comment by andyw8** [comment]
> This is due to the gem requiring ActiveResource < 6.0.0 (so it's using 5.1.1).

5.1.1 is the latest release of ActiveResource, but there are some unreleased commits in master relating to `URLParser`. I'll see if we can get an updated release.

**Comment by andyw8** [comment]
@key88sf in the meantime, you can try pointing your Gemfile to the master branch of ActiveResource to confirm that eliminates the warnings.

**Comment by noctivityinc** [comment]
Any update on this?  It's been almost two months.  Is shopify_api going to be updated?

**Comment by DannyBen** [comment]
This happens in a non rails environment as well - perhaps the title of the issue should be changed?

This is the simplest reproduction code

```ruby
require 'shopify_api'
ShopifyAPI::Base.site = "https://key:secret@app.myshopify.com/admin"
```

Running this code gives the below error (twice for some reason):

```
DEPRECATION WARNING: URI.parser is deprecated and will be removed in Rails 6.2. Use `URI::DEFAULT_PARSER` instead. (called from <main> at test.rb:2)
```

Note that when given a simpler URL (without credentials), the warning is not triggered, in case it helps debugging.

**Comment by andyw8** [comment]
The warning comes from ActiveSupport which a dependency of ActiveResource, which this gem uses.

**Comment by andyw8** [comment]
We are waiting on a new release of ActiveResource: https://github.com/rails/activeresource/issues/350

**Comment by rsnodgrass** [comment]
This is still an issue, and now activeresource/activesupport/etc are released 7.0 as of December 15 which actually breaks shopify_api from working at all.

**Comment by cornexo** [comment]
@andyw8 activeresource was updated to 6.0.0 today which includes this fix but also drops support for Ruby < 2.6 and Rails < 6.0.
